### DISCUSSION TITLE
trial inspection of fits

### DISCUSSION THREAD

**Discussion by BePhysLab-M** [bachlab/PsPM#765] [Q&A] [answered]
### Summary
Model predictions drop and oscillations
### Technical Info
* PsPM revision ID:6.0.0
* MATLAB version:R2024a

Apologies beforehand if the following question is rather daft!

I am trying to help someone out regarding a fear conditioning experiment. Measurement is EDA, stimulation is electrical, other cues visual. After running a DCM (default settings), the trial by trial inspection plot (for one model/participant) shows a sudden drop in model predictions for the last trial and oscillations on a few others.

Would it be possible to provide a pointer as to why these things (could) occur?
In attach a snip, it's present in the top and bottom trials.

Many thanks in advance!

![Screenshot 2024-08-14 165341 _send PSPM github](https://github.com/user-attachments/assets/911d3162-bc2c-4055-9f6f-c470aa7debdb)

**Comment by dominikbach** [comment]
The oscillations are likely due to the fact that the data are sustained above zero (while the model predicts a decay towards zero). DCM subtracts the minimum data value from the data (as an implicit baseline) but this does not seem to be the case here. I wonder whether something is wrong with the data (typically, due to brief artefacts). Could you share a sample file?

Regarding the sudden drop, could you share the model, so that we can investigate?

**Comment by BePhysLab-M** [comment]
Dear Dominik,



Excellent, thank you for your speedy response!


The data were recorded using a Biopac MP160, EDA100D smart amp.
Exported to .mat from Acqknowledge.
Downsampled from 2 to 1 Khz.
Trimmed 30 seconds before the first and 30 seconds after the last marker (done in batch to avoid artefacts due to electrodes still being attached etcet).
Electrical stimulation is a pulse train using a Digitimer DS7, was briefly wondering whether that could've been misaligned somehow, but seems unlikely.

[cid:a9a7a584-96a0-4cf1-bf46-1c2e86b84a2a]


I did not spot any major artefacts (electrode disconnects, what is noted in the video lecture to be potentially problematic).

Could the missing prediction be related to the Last trial cutoff [s] (default: 7s) parameter?

It should be possible to download the files (model and data; data is anonymous) using this link: share<https://kuleuven-my.sharepoint.com/:f:/g/personal/mathijs_franssen_kuleuven_be/ErsgY_v_sM5Gll-v74PinVwBGUtbsYxHyhxbMfxSOoCm8A?e=OCAONU>
Password is 1685

Apologies for the delay in responding!


Many thanks in advance!

All the best

Mathijs


________________________________

**Comment by dominikbach** [comment, marked as answer]
Hi Mathijs

(1) Sudden drop of predicted signal in the last trial: PsPM uses only the ten seconds after last trial end to estimate responses for that trial. This is a somewhat arbitrary choice which we could easily change, although in your case the last response appears to be fit quite well, so nothing to worry about.

(2) With your relatively long ITIs, the bidirectional filter creates ringing artefacts that are not well fit by the model, and may explain the "oscillations" between trials that you observe. We have seen this effect before (https://doi.org/10.1111/psyp.14119) and have used unidirectional filtering in this case. We are currently working on a more systematic investigation and are probably going to change the default to unidirectional filtering in the future. You may want to try this.

Dominik
